typhon/utils.py

```diff
@@ -31,6 +32,57 @@
 ui_dir = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'ui')
 
 
+def UsesEnums(*enum_classes, base=QWidget):
+    '''
+    Add Qt Designer compatible enums to a class. Under the hood, this generates
+    a base class for the target class.
+
+    Parameters
+    ----------
+    *enum_classes : enum.Enum or namespace
+        Enum classes - can be either of type `enum.Enum` or a simple namespace
+        where one attribute is associated with one value.
+    base : class
+        The base class for the generated Qt-compatible class. Must be either
+        QObject or QWidget (or have one in the class hierarchy)
+    '''
+    class EnumBase(base):
+        for enum in enum_classes:
+            Q_ENUMS(enum)
+
+        _qt_enums = enum_classes
+
+        @classmethod
+        def _find_qt_enums(class_):
```

Comment:
This doesn't get called anywhere? What is its purpose?

Reply:
To `Return the set of all used Q_ENUMS on this class`
It just adds extra introspection capabilities to anything using `EnumBase`. If not deemed useful, both `_qt_enums` and `_find_qt_enums` could be nuked.

Reply:
I think maybe keep `_qt_enums` but drop the discovery function? Not sure what the advantage is of having both, but maybe you could explain in person.